ndk-macro/src/helper.rs

```diff
@@ -49,16 +49,11 @@ fn crate_name(name: &str) -> Result<String> {
 pub fn crate_path(name: &str, overridden_path: &Option<Path>) -> Path {
     // try to use overridden crate path
     overridden_path.clone().unwrap_or_else(|| {
-        // the binding to hold string from `crate_name` fn
-        let mut detected_name = None;
         Ident::new(
             // try to determine crate name from Cargo.toml
             crate_name(name)
                 .ok()
-                .map(|name| {
-                    detected_name = Some(name);
```

Comment:
what's this :laughing:

Reply:
:joy: I think `.map(|name| name.as_str())` was tried first which moves the `String` from the `Option<>` into the closure, resulting in obvious lifetime issues when calling `.as_str()`. The way to solve that is to use `.as_ref().map(|name| ...)`, and in this case an even simpler `.as_deref()` because `String`'s `Deref` is implemented to target `str` :grimacing: (since the `String` stays alive during this entire statement).

But you knew that already, just writing it here for others who might stumble across :grinning: